src/Auth0.php

```diff
@@ -269,6 +268,7 @@ public function exchange(
         }
 
         if ($state === null || ! $this->getTransientStore()->verify('state', $state)) {
+            $this->deferStateSaving(false);
```

Comment:
Why are writing the cookie and then immediately clearing it? Question applies to the other changes below where we write then clear.

Reply:
I've folded these extraneous into clear() so that the write only occurs once after the local state is cleared. Thanks for spotting this opportunity for further optimization!